apiserver/facades/client/bundle/bundle.go

```diff
@@ -248,8 +248,13 @@ func (b *BundleAPI) fillBundleData(model description.Model) (*charm.BundleData,
 			Annotations: machine.Annotations(),
 			Series:      machine.Series(),
 		}
-		if result := b.constraints(machine.Constraints()); len(result) != 0 {
+
+		if result := b.hardwareConstraints(machine.Instance()); len(result) != 0 {
 			newMachine.Constraints = strings.Join(result, " ")
+		} else {
+			if result = b.machineConstraints(machine.Constraints()); len(result) != 0 {
```

Comment:
We don't need this here right?
Since we set machine constraints earlier up above?

Reply:
Oh wait, I got confused because of the method name. Because it's also used above to get the application constraints, maybe we should revert the name back to "constraints" and have the hardware constraints version called hardwareConstraints is is the case now.

Reply:
:) ok.